lib/kafo/puppet_module.rb

```diff
@@ -12,7 +12,11 @@ class PuppetModule
     attr_reader :name, :identifier, :params, :dir_name, :class_name, :manifest_name, :manifest_path,
                 :groups, :params_path, :params_class_name, :configuration, :raw_data
 
-    def initialize(identifier, parser = KafoParsers::PuppetModuleParser, configuration = KafoConfigure.config)
+    def self.find_parser
+      @@parser ||= KafoParsers::Parsers.find_available(:logger => KafoConfigure.logger) || :none
```

Comment:
But this is a class method, so instance scope is PuppetModule object scope. The difference is when some class would inherit from PuppetModule, so the class variable would be shared with children which I don't think is expected.

```ruby
class A
  def self.a
    @a ||= rand(1000)
  end
end

10.times {
  puts A.new.class.a # will always print the same number
}
```